superset/reports/notifications/base.py

```diff
@@ -28,7 +28,8 @@
 class NotificationContent:
     name: str
     header_data: HeaderDataType  # this is optional to account for error states
-    csv: Optional[bytes] = None  # bytes for csv file
+    data: Optional[bytes] = None  # bytes for data attachment
+    data_format: Optional[str] = None  # data attachment format (csv, xlsx, pdf, etc)
```

Comment:
I think we should use an `Enum` or `Literal` as the type here to make it detectable by the linter if a `data_format` is valid.